repository-postgresql/src/main/java/org/lfenergy/compas/scl/data/repository/postgresql/CompasSclDataPostgreSQLRepository.java

```diff
@@ -155,6 +155,29 @@ public String findByUUID(SclFileType type, UUID id, Version version) {
         }
     }
 
+    @Override
+    public boolean hasDuplicateSclName(SclFileType type, String name) {
+        var sql = "SELECT DISTINCT ON (id) m.* "
+                + "FROM scl_file m "
```

Comment:
There constants for parts of the query to prevent duplicate code messages in the future.

Reply:
`FROM_CLAUSE` can be re used here indeed :)